### PR TITLE
Fix for #775: Iterable forged methods and lower bounds

### PR DESCRIPTION
**Pull request by agudian**
#775 actually uncovers two problems, I tried to fix them in two separate commits. The creation of the forged method for Iterables is straight forward I guess, but the lower-bound matching is a bit voodooish... I didn't get it to break though while trying around a bit. We still might have similar issues with upper-bounds, in case anyone wants to try that out in detail... :smirk:

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_775/IterableContainer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.bugs._775;
+
+/**
+ * @author Andreas Gudian
+ */
+public class IterableContainer {
+    private Iterable<? extends Integer> values;
```
Comment (gunnarmorling) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_775/IterableContainer.java]: LOL, `Integer` is final, but I guess for the sake of the test... ;)

**Comment by gunnarmorling** [comment]
LGTM, I'll merge once the enum stuff is in.

**Comment by gunnarmorling** [comment]
> We still might have similar issues with upper-bounds, in case anyone wants to try that out in detail...

Ok, I'd say let's take these as they are actually reported. It's a continuous process here :)

**Comment by sjaakd** [comment]
> LGTM, I'll merge once the enum stuff is in.

Its in...

**Pull request merged**